scripts/check_release_branch_date.py

```diff
@@ -0,0 +1,22 @@
+import json
+from datetime import datetime, date, timedelta
+from urllib import request
+
+from common import get_patch_version, env
+
+if __name__ == '__main__':
+    repo = env("GITHUB_REPOSITORY")
+    response = request.urlopen(f"https://api.github.com/repos/{repo}/milestones")
+    milestones = json.load(response)
+
+    milestone_version = f"v{get_patch_version()}"
+    milestone = next(milestone for milestone in milestones if milestone["title"] == milestone_version)
```

Comment:
Some proper error handling here can be helpful, if no milestones were found it will fail with `StopIteration` exception. The step itself will "succeed" and the next step will fail with some cryptic "The template is not valid" error. See [my example](https://github.com/lancelote/intellij-rust/runs/994205961?check_suite_focus=true) when I accidentally run the workflow against my fork with no milestones available. Though it is not crucial.